atoma-service/src/handlers/mod.rs

```diff
@@ -1045,56 +727,21 @@ pub mod inference_service_metrics {
             .cloned()
             .partition(|(_, job)| job.contains("vllm"));
 
-        let mut min_time_to_first_token_seconds = f64::INFINITY;
-
         tracing::debug!(
             target = "atoma-service",
             module = "inference_service_metrics",
             level = "info",
             "Partitioned chat completions service urls: vllm: {vllm_chat_completions_service_urls:?}, sglang: {sglang_chat_completions_service_urls:?}"
         );
 
-        // Get cached metrics
-        let vllm_metrics = if vllm_chat_completions_service_urls.is_empty() {
-            vec![]
-        } else if let Some(metrics) = VLLM_METRICS_CACHE.get_metrics().await {
-            metrics
-        } else {
-            info!(
-                target = "atoma-service",
-                module = "inference_service_metrics",
-                level = "info",
-                "No cached vLLM metrics, getting them directly"
-            );
-            let vllm_chat_completions_service_urls_with_model: Vec<(String, String, String)> =
-                vllm_chat_completions_service_urls
-                    .iter()
-                    .map(|(url, job)| (model.to_string(), url.clone(), job.clone()))
-                    .collect();
-            get_metrics_vllm(&HTTP_CLIENT, &vllm_chat_completions_service_urls_with_model).await
-        };
-        let sglang_metrics = if sglang_chat_completions_service_urls.is_empty() {
-            vec![]
-        } else if let Some(metrics) = SGLANG_METRICS_CACHE.get_metrics().await {
-            metrics
-        } else {
-            info!(
-                target = "atoma-service",
-                module = "inference_service_metrics",
-                level = "info",
-                "No cached SgLang metrics, getting them directly"
-            );
-            let sglang_chat_completions_service_urls_with_model: Vec<(String, String, String)> =
-                sglang_chat_completions_service_urls
-                    .iter()
-                    .map(|(url, job)| (model.to_string(), url.clone(), job.clone()))
-                    .collect();
-            get_metrics_sglang(
-                &HTTP_CLIENT,
-                &sglang_chat_completions_service_urls_with_model,
-            )
-            .await
-        };
+        let vllm_metrics =
+            get_metrics(InferenceService::Vllm, &vllm_chat_completions_service_urls).await;
+        let sglang_metrics = get_metrics(
```

Comment:
This doesn't make sense. Either you extract vLLM or SgLang, or both if it is really the case both services are being used, but you should check

Reply:
I disagree, this will just polute code if unnecessary check.

Reply:
I disagree it is poluting the code. I would rather not have unnecessary checks and unnecessary logs that just makes life more difficult for who is operating the node logic.

Reply:
so you want to add check if the vector e.g. `vllm_chat_completions_service_urls` is empty, and not call `get_metrics` if it's empty? but calling `get_metrics` with an empty array just does something like this `vec![].iter().collect()` so just returns empty vec